### PR TITLE
Tad can no longer perform instant transmission

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Closes https://github.com/tgstation/TerraGov-Marine-Corps/issues/14477
Adds 1 safety check (Tad cannot land on its own z level)
+ Actually closes the UI and boots you out of camera view.
## Why It's Good For The Game
Closes https://github.com/tgstation/TerraGov-Marine-Corps/issues/14477
Bugfix
## Changelog
:cl:
fix: Tad can no longer land on itself
fix: Tad can no longer perform instant transmission
/:cl:

### CODE REVIEW

File: code/modules/shuttle/mini_dropship.dm [hunk 1]
```diff
@@ -279,10 +279,15 @@
 	if(!origin.placeLandingSpot(target))
 		to_chat(owner, span_warning("You cannot land here."))
 		return
+	if(is_ground_level(origin.z)) //Safety check to prevent instant transmission
+		to_chat(owner, span_warning("The shuttle can't move while docked on the planet"))
+		return
 	origin.shuttle_port.callTime = SHUTTLE_LANDING_CALLTIME
 	origin.next_fly_state = SHUTTLE_ON_GROUND
 	origin.open_prompt = FALSE
-	origin.clean_ui_user()
```
Comment (ivanmixo) [inline code/modules/shuttle/mini_dropship.dm:285]: Just... why not do these 3 lines below in clean_ui_user itself?

**Review by TiviPlus — APPROVED** [review]

File: code/modules/shuttle/mini_dropship.dm [hunk 2]
```diff
@@ -279,10 +280,13 @@
...
+	origin.ui_close(C)
```
Comment (TiviPlus) [inline code/modules/shuttle/mini_dropship.dm:289]:
```suggestion
	SStgui.close_user_uis(C, origin)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/mini_dropship.dm (hunk 2 above)
Comment (Xander3359) [reply code/modules/shuttle/mini_dropship.dm:289]: That line gets ran in the proc

Comment (ivanmixo) [reply code/modules/shuttle/mini_dropship.dm:289]: Again, so why is this line necessary at all then?

Comment (TiviPlus) [reply code/modules/shuttle/mini_dropship.dm:289]: `/**
 * public
 *
 * Called on a UI's object when the UI is closed, not to be confused with
 * client/verb/uiclose(), which closes the ui window
 */
/datum/proc/ui_close(mob/user)
	SIGNAL_HANDLER
`

**Pull request merged**